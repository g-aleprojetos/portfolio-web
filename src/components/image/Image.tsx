import React from 'react';
import {ImageStyled, Svg, Props} from './Image.styles';

interface IconeProps extends Props {
  testId?: string;
  src: string;
  alt?: string;
  cor?: string;
  spinner?: boolean;
  velocidadeSpinner?: number;
}

export const Image = ({
  testId,
  src,
  alt,
  cursor,
  altura = 32,
  largura = 32,
  inclinacao = 0,
  spinner = false,
  velocidadeSpinner = 0,
  cor,
  ...props
}: Readonly<IconeProps>) => {
  const isSvg = src.endsWith('.svg');

  return (
    <>
      {isSvg ? (
        <Svg
          data-testid={testId ? `${testId}-svg` : 'svg'}
          src={src}
          cor={cor}
          width={largura}
          height={altura}
          cursor={cursor}
          $inclinacao={inclinacao}
          $spinner={spinner}
          $velocidadeSpinner={velocidadeSpinner}
        />
      ) : (
        <ImageStyled
          data-testid={testId ? `${testId}-img` : 'img'}
          src={src}
          alt={alt}
          width={largura}
          height={altura}
          cursor={cursor}
          $inclinacao={inclinacao}
          $spinner={spinner}
          $velocidadeSpinner={velocidadeSpinner}
          {...props}
        />
      )}
    </>
  );
};
