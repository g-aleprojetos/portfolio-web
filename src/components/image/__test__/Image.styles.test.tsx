import React from 'react';
import {render, act} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import {ImageStyled, Svg, ContainerImage} from '../Image.styles';

describe('Image.styles', () => {
  test('ImageStyled DEVE ser igual ao snapshot', async () => {
    await act(async () => {
      const {container} = render(<ImageStyled />);
      expect(container).toMatchSnapshot();
    });
  });

  test('ImageStyled DEVE ser igual ao snapshot QUANDO a inclinacao 45', async () => {
    await act(async () => {
      const {container} = render(<ImageStyled $inclinacao={45} />);
      expect(container).toMatchSnapshot();
    });
  });

  test('ImageStyled DEVE ser igual ao snapshot QUANDO o cursor for pointer', async () => {
    await act(async () => {
      const {container} = render(<ImageStyled cursor="pointer" />);
      expect(container).toMatchSnapshot();
    });
  });

  test('ImageStyled DEVE ser igual ao snapshot QUANDO o cursor for default', async () => {
    await act(async () => {
      const {container} = render(<ImageStyled cursor="default" />);
      expect(container).toMatchSnapshot();
    });
  });

  test('ImageStyled DEVE ser igual ao snapshot QUANDO o spinner for true e a velocidadeSpinner for 2', async () => {
    await act(async () => {
      const {container} = render(
        <ImageStyled $spinner={true} $velocidadeSpinner={2} />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  test('ContainerImage DEVE ser igual ao snapshot QUANDO largura e altura são fornecidas como strings', async () => {
    await act(async () => {
      const {container} = render(
        <ContainerImage $largura="100px" $altura="100px" />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  test('ContainerImage DEVE ser igual ao snapshot QUANDO apenas largura é fornecida como string', async () => {
    await act(async () => {
      const {container} = render(<ContainerImage $largura="150px" />);
      expect(container).toMatchSnapshot();
    });
  });

  test('ContainerImage DEVE ser igual ao snapshot QUANDO apenas altura é fornecida como string', async () => {
    await act(async () => {
      const {container} = render(<ContainerImage $altura="200px" />);
      expect(container).toMatchSnapshot();
    });
  });

  test('Svg DEVE ser igual ao snapshot', async () => {
    await act(async () => {
      const {container} = render(<Svg src="path/to/your.svg" />);
      expect(container).toMatchSnapshot();
    });
  });

  test('Svg DEVE ser igual ao snapshot QUANDO a inclinacao 45', async () => {
    await act(async () => {
      const {container} = render(
        <Svg src="path/to/your.svg" $inclinacao={45} />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  test('Svg DEVE ser igual ao snapshot QUANDO o cursor for pointer', async () => {
    await act(async () => {
      const {container} = render(
        <Svg src="path/to/your.svg" cursor="pointer" />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  test('Svg DEVE ser igual ao snapshot QUANDO o cursor for default', async () => {
    await act(async () => {
      const {container} = render(
        <Svg src="path/to/your.svg" cursor="default" />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  test('Svg DEVE ser igual ao snapshot QUANDO o spinner for true e a velocidadeSpinner for 2', async () => {
    await act(async () => {
      const {container} = render(
        <Svg src="path/to/your.svg" $spinner={true} $velocidadeSpinner={2} />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  test('Svg DEVE ser igual ao snapshot QUANDO a cor for fornecida', async () => {
    await act(async () => {
      const {container} = render(<Svg src="path/to/your.svg" color="red" />);
      expect(container).toMatchSnapshot();
    });
  });
});
