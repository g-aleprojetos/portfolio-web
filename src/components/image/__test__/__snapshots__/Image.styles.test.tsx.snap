// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image.styles ImageStyled DEVE ser igual ao snapshot 1`] = `
.c0 {
  pointer-events: auto;
  cursor: default;
  transform: rotate(deg);
  width: px;
  height: px;
}

<div>
  <img
    class="c0"
  />
</div>
`;

exports[`Image.styles ImageStyled DEVE ser igual ao snapshot QUANDO a inclinacao 45 1`] = `
.c0 {
  pointer-events: auto;
  cursor: default;
  transform: rotate(45deg);
  width: px;
  height: px;
}

<div>
  <img
    class="c0"
  />
</div>
`;

exports[`Image.styles ImageStyled DEVE ser igual ao snapshot QUANDO o cursor for default 1`] = `
.c0 {
  pointer-events: none;
  cursor: default;
  transform: rotate(deg);
  width: px;
  height: px;
}

<div>
  <img
    class="c0"
    cursor="default"
  />
</div>
`;

exports[`Image.styles ImageStyled DEVE ser igual ao snapshot QUANDO o cursor for pointer 1`] = `
.c0 {
  pointer-events: auto;
  cursor: pointer;
  transform: rotate(deg);
  width: px;
  height: px;
}

<div>
  <img
    class="c0"
    cursor="pointer"
  />
</div>
`;

exports[`Image.styles ImageStyled DEVE ser igual ao snapshot QUANDO o spinner for true e a velocidadeSpinner for 2 1`] = `
.c0 {
  pointer-events: auto;
  cursor: default;
  transform: rotate(deg);
  width: px;
  height: px;
  animation: fvtopB infinite 2s linear;
}

<div>
  <img
    class="c0"
  />
</div>
`;

exports[`Image.styles Svg DEVE ser igual ao snapshot 1`] = `<div />`;

exports[`Image.styles Svg DEVE ser igual ao snapshot QUANDO a inclinacao 45 1`] = `<div />`;

exports[`Image.styles Svg DEVE ser igual ao snapshot QUANDO o cursor for default 1`] = `<div />`;

exports[`Image.styles Svg DEVE ser igual ao snapshot QUANDO o cursor for pointer 1`] = `<div />`;

exports[`Image.styles Svg DEVE ser igual ao snapshot QUANDO o spinner for true e a velocidadeSpinner for 2 1`] = `<div />`;
